src/simsopt/_core/graph_optimizable.py

```diff
@@ -1295,3 +1315,57 @@ def J(self):
             return self.func(*args, **kwargs)
 
     return TempOptimizable(func, *args, dof_indicators=dof_indicators, **kwargs)
+
+
+class ScaledOptimizable(Optimizable):
+    """
+    Represents an :obj:`~simsopt._core.graph_optimizable.Optimizable`
+    object scaled by a constant factor. This class is useful for
+    including a weight in front of terms in an objective function. For
+    now, this feature works on classes for which ``.J()`` returns an
+    objective value and ``.dJ()`` returns the gradient, e.g. coil
+    optimization.
+
+    Args:
+        factor: (float) The constant scale factor.
+        opt: An :obj:`~simsopt._core.graph_optimizable.Optimizable` object to scale.
+    """
+
+    def __init__(self, factor, opt):
+        self.factor = factor
+        self.opt = opt
+        super().__init__(depends_on=[opt])
+
+    def J(self):
+        return self.factor * self.opt.J()
+
+    @derivative_dec
+    def dJ(self):
+        # Next line uses __rmul__ function for the Derivative class
+        return self.factor * self.opt.dJ(partials=True)
+
+
+class OptimizableSum(Optimizable):
+    """
+    Represents a sum of
+    :obj:`~simsopt._core.graph_optimizable.Optimizable` objects. This
+    class is useful for combining terms in an objective function. For
+    now, this feature works on classes for which ``.J()`` returns an
+    objective value and ``.dJ()`` returns the gradient, e.g. coil
+    optimization.
+
+    Args:
+        opts: A python list of :obj:`~simsopt._core.graph_optimizable.Optimizable` object to sum.
+    """
+
+    def __init__(self, opts):
```

Comment:
I like being able to add together an arbitrary number (>2) of objects since this is a common use case. This occurs for instance when adding together the coil length objectives for all the coils, or when adding together all the quadratic flux objectives for the stochastic sample average in stochastic optimization.

Reply:
OK